fad-win32.c

```diff
@@ -42,6 +42,107 @@
 
 #include <errno.h>
 
+/*
+* Add an entry to the list of addresses for an interface.
+* "curdev" is the entry for that interface.
+*/
+static int
```

Comment:
Why is this routine necessary?  What does it do that add_addr_to_dev() doesn't do (other than depend on struct sockaddr_storage existing)?  There are some things add_addr_to_dev() does that this routine does *NOT* do, such as:

- properly free previously-allocated memory if a later allocation fails;

- use `pcap_snprintf()` rather than `snprintf()`.

Furthermore, it doesn't have the size arguments, but your later call to it *does* pass the size arguments, so the code won't even compile.